### PR TITLE
add components version verification

### PR DESCRIPTION
**Pull request by DocentSzachista**
Signed-off-by: Damian Raczkowski damian.raczkowski@intel.com

### CODE REVIEW

File: test/functional/tests/ci/test_basic.py [hunk 1]
```diff
@@ -0,0 +1,36 @@
...
+    file = File(directory)
+    content = file.read().split("\n")
+    file_cas_version = ""
+    for line in content:
+        if "CAS_VERSION=" in line:
+            file_cas_version = line.split('=')[1]
+            break
+    file_cas_version = CasVersion.from_version_string(file_cas_version)
+    for version in cmd_cas_versions:
+        assert version.split(",")[1] == str(file_cas_version)
```
Comment (karolinavelkaja) [inline test/functional/tests/ci/test_basic.py:36]: here as well (and in general) :)

File: test/functional/tests/ci/test_basic.py [hunk 2]
```diff
@@ -0,0 +1,36 @@
...
+from api.cas.version import CasVersion
+from test_utils.filesystem.file import File
+
+
+def test_if_CAS_exists():
+    """
+        title: Check if CAS is installed
+        description: Check if cAS is installed with `--version` flag
+        pass_criteria:
+          - Command returns exit_code 0
```
Comment (pdebski21) [inline test/functional/tests/ci/test_basic.py:18]: casadm command succeeds is more explicit :)

File: test/functional/tests/ci/test_basic.py [hunk 3]
```diff
@@ -0,0 +1,36 @@
...
+
+from core.test_run import TestRun
+from api.cas.version import CasVersion
+from test_utils.filesystem.file import File
+
+
+def test_if_CAS_exists():
+    """
+        title: Check if CAS is installed
+        description: Check if cAS is installed with `--version` flag
```
Comment (pdebski21) [inline test/functional/tests/ci/test_basic.py:16]: misspelling cAS

File: test/functional/tests/ci/test_basic.py [hunk 4]
```diff
@@ -0,0 +1,36 @@
...
+        pass_criteria:
+          - Command returns exit_code 0
+          - Versions are matched from cmd and file in /var/lib/opencas/cas_version
+    """
+    cmd = f"casadm --version -o csv"
+    output = TestRun.executor.run(cmd)
+    assert output.exit_code == 0
+    cmd_cas_versions = output.stdout.split("\n")[1:]
+
+    directory = r"/var/lib/opencas/cas_version"
```
Comment (pdebski21) [inline test/functional/tests/ci/test_basic.py:26]: Little misleading variable name. Suggest `version_file` or `version_file_path`.

**Comment by DocentSzachista** [comment]
I have pushed changes that you proposed. Please take a look :)

File: test/functional/tests/ci/test_basic.py [hunk 5]
```diff
@@ -0,0 +1,39 @@
...
+
+def test_cas_version():
+    """
+        title: Check if CAS is installed
+        description: Check if CAS is installed with `--version` flag
+        and later components version is compared with version file
+        pass_criteria:
+          - `casadm` command succeeds
+          - Versions are matched from cmd and file in /var/lib/opencas/cas_version
+    """
```
Comment (pdebski21) [inline test/functional/tests/ci/test_basic.py:21]: Please align with Test description format as in other tests.
Also leave --version and casadm without " ` " and pass_criteria without "_".

Comment (DocentSzachista) [reply test/functional/tests/ci/test_basic.py]: Done :)

**Review by pdebski21 — DISMISSED** [review]

File: test/functional/tests/ci/test_basic.py [hunk 6]
```diff
@@ -0,0 +1,40 @@
...
+      Check if CAS is installed with --version flag and later
+      checks if components version is consistent with version file
+    pass criteria:
+     - casadm command succeeds
+     - Versions are matched from cmd and file in /var/lib/opencas/cas_version
+    """
+    cmd = f"casadm --version -o csv"
+    output = TestRun.executor.run(cmd)
+    if output.exit_code != 0:
+        TestRun.LOGGER.error(f"Command should return version")
```
Comment (mmichal10) [inline test/functional/tests/ci/test_basic.py:26]: You can use `TestRun.executor.run_expect_success(cmd)`. It's more concise and does exactly the same thing

File: test/functional/tests/ci/test_basic.py [hunk 7]
```diff
@@ -0,0 +1,40 @@
...
+    cmd_cas_versions = output.stdout.split("\n")[1:]
+
+    version_file_path = r"/var/lib/opencas/cas_version"
+    file = File(version_file_path)
+    content = file.read().split("\n")
+    file_cas_version = ""
+    for line in content:
+        if "CAS_VERSION=" in line:
+            file_cas_version = line.split('=')[1]
+            break
```
[the remark this answers is not in the document]
Comment (DocentSzachista) [reply test/functional/tests/ci/test_basic.py]: Thanks for the insight :)

File: test/functional/tests/ci/test_basic.py [hunk 8]
```diff
@@ -0,0 +1,40 @@
...
+    file_cas_version = CasVersion.from_version_string(file_cas_version)
+    for version in cmd_cas_versions:
+        if version.split(",")[1] != str(file_cas_version):
+            TestRun.LOGGER.error(f"versions from cmd and file doesn't match")
```
[the remark this answers is not in the document]
Comment (DocentSzachista) [reply test/functional/tests/ci/test_basic.py]: Done :)

File: test/functional/tests/ci/test_basic.py (hunk 6 above)
Comment (DocentSzachista) [reply test/functional/tests/ci/test_basic.py]: Done :)

**Review by mmichal10 — APPROVED** [review]

**Review by pdebski21 — APPROVED** [review]

**Comment by mmichal10** [comment]
Please acknowledge in some way that this commit contains a new test. E.g. you can change test commit message to `tests: add components version verification`

**Review by mmichal10 — APPROVED** [review]

**Review by pdebski21 — APPROVED** [review]

**Review by KlaudiaJ — APPROVED** [review]

**Review by pdebski21 — APPROVED** [review]

**Review by karolinavelkaja — APPROVED** [review]

**Comment by mmichal10** [comment]
retest this please

**Pull request merged**